### PR TITLE
Update GPS platforms and group GNSSRO data referring to "record_number".

### PR DESCRIPTION
**Pull request by gmao-jjin3**
Add more GPS platforms from which data are used in GEOS. Convert "Errinv_Input" to "GsiInputObsError" for all of the types of observations. However, Errinv_Input is removed because a PR issue (Nov 18, 2022).

There are regroup/sorting sections which assign data with same "record_number" together for GNSSRO bending angle observations.  

It solves the issue in  https://github.com/JCSDA-internal/ioda-converters/issues/1072 .

### CODE REVIEW

**Review by danholdaway — APPROVED** [review]

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by BenjaminRuston** [comment]
@gmao-jjin3 any idea what's going on with this failed test:

`test_iodaconv_gsidiag_conv (Failed)`

**Comment by CoryMartin-NOAA** [comment]
> @gmao-jjin3 any idea what's going on with this failed test:
> 
> `test_iodaconv_gsidiag_conv (Failed)`

My guess is that the test reference output files need updated since the input error was added

**Comment by BenjaminRuston** [comment]
@gmao-jjin3 .... See you followed what @CoryMartin-NOAA said and updated the `testoutput`

can you replicate this error?  

`

Start  56: test_iodaconv_gsidiag_conv
--
553 | 56/249 Test  #56: test_iodaconv_gsidiag_conv ......................................................***Failed    0.31 sec

`

**Comment by gmao-jjin3** [comment]
GNSSRO bending angle data saved from GSI need to be grouped for each "record_number".  See the @gmao-yzhu 's post in https://github.com/GEOS-ESM/swell/issues/67#issuecomment-1320125496 on Nov 16, 2022.  This update first sorts/re-groups the record_number of bending angle data and saves other variables referring to subscripts of "record_number".  GSI hofx can be reproduced by products of this update, which is tested by @gmao-yzhu. 

The original "sfc_tv_obs_2018041500.nc4" is brought back after the newly added "Errinv_Input" is taken out. It seems there is no way for this PR to pass the on-line test even it passes the local test.

**Comment by gmao-jjin3** [comment]
@CoryMartin-NOAA  Can you take a look at those new changes and merge them?  @danholdaway

File: src/gsi-ncdiag/gsi_ncdiag.py [hunk 1]
```diff
@@ -187,6 +189,7 @@
 gsi_add_vars = {
     'ObsBias': 'GsiObsBias',
     'Observation_Type': 'ObsType',
+    'Observation_Subtype': 'ObsSubType',
```
Comment (CoryMartin-NOAA) [inline src/gsi-ncdiag/gsi_ncdiag.py:192]: Just a question, does this follow the convention that we will change to in 2 weeks time?

Comment (gmao-jjin3) [reply src/gsi-ncdiag/gsi_ncdiag.py:192]: Not sure. I have no knowledge about that.  We need those subtypes for our configuration.

Comment (gthompsnJCSDA) [reply src/gsi-ncdiag/gsi_ncdiag.py:192]: This is a variable name I placed into the `ObsSpace.yaml` file that we will be using soon to validate that incoming data has all the variable names to match the JEDI Data Conventions.  Please use this: `observationSubTypeNum`

**Review by BenjaminRuston — APPROVED** [review]
@gmao-jjin3 we will go ahead and merge this for now but realize by the 17Jan this will need to update.  @gthompsnJCSDA will take a crack at it in the sprint branch and will inform @danholdaway of this again on Friday 06Jan.

**Comment by gthompsnJCSDA** [comment]
@gmao-jjin3   I made all the necessary changes to the sprint branch to incorporate all your changes plus all the naming convention changes - so there should be no further work needed on this code.

**Comment by gmao-jjin3** [comment]
@gthompsnJCSDA @BenjaminRuston  Many thanks!

**Pull request merged**